### PR TITLE
fix: refactor to make filterIgnoredIssues match vuln ids case insensitively

### PR DESCRIPTION
**Pull request by pmlyng**
#### What does this PR do?

Ignored issues are not filtered out case insensitively. 
This adds a test and updates the logic to match the vulnID (named as issue?.issueId or as the key of the issuesData object), with the policy file's ignore object's keys in a case-insensitive manner.

#### Where should the reviewer start?
~~This is a draft PR~~

#### Any background context you want to provide?
Originally we thought this could be resolved by bumping the version of the policy lib (https://github.com/snyk/cli/pull/4731), however through acceptance testing, we found that the logic here means filtering is still case-sensitive so that wouldn't completely solve the problem.

~~This is still a draft PR, but when ready, we can cherry-pick the changes from this PR into #4731.~~

cherry-picked the changes from #4731 into this one 

#### What are the relevant tickets?
Support ticket here: https://snyksec.atlassian.net/browse/NARW-2259

### CODE REVIEW

**Comment by darscan** [comment]
👋 This change introduces a _lot_ more processing, and tight nested loops into a synchronous operation. Are you sure there isn't a neater and cheaper implementation?

File: src/lib/ecosystems/policy.ts [hunk 1]
```diff
@@ -42,23 +42,48 @@ export function filterIgnoredIssues(
...
 
-  const filteredIssuesData: IssuesData = { ...issuesData };
+  const upperCaseKeys = (obj) => {
+    return Object.keys(obj).reduce((acc, k) => {
+      acc[k.toUpperCase()] = obj[k];
+      return acc;
+    }, {});
+  };
+
+  const policyIgnoreKeyUpperCased = upperCaseKeys({ ...policy.ignore });
```
Comment (gclapperton) [inline src/lib/ecosystems/policy.ts:52]: Should we normalize the keys in the policy lib?

**Review by YairZ101 — APPROVED** [review]
Approving for cloud-dev-ex

**Review by jcsackett — APPROVED** [review]

**Comment by pmlyng** [comment]
Closing for now as summarised beautifully by @jcsackett  here https://snyk.slack.com/archives/CL6DF1CDT/p1694098802648089

**Pull request closed without merging**